mergify_engine/context.py

```diff
@@ -752,6 +753,66 @@ async def retrieve_unverified_commits(self) -> typing.List[str]:
             if not commit["commit_verification_verified"]
         ]
 
+    async def retrieve_resolved_review_threads(
+        self, resolved: typing.Optional[bool] = True
+    ) -> ContextAttributeType:
+        query = """
+            repository(owner: "{owner}", name: "{name}") {{
+                pullRequest(number: {number}) {{
+                    reviewThreads(first: 100{after}) {{
+                    edges {{
+                        node {{
+                        isResolved
+                        line
+                        path
+                        comments(first: 1) {{
+                            edges {{
+                                node {{
+                                    body
+                                }}
+                            }}
+                        }}
+                        }}
+                    }}
+                    }}
+                }}
+            }}
+        """
+        responses = multi.multi_query(
+            query,
+            iterable=(
+                {
+                    "owner": self.repository.repo["owner"]["login"],
+                    "name": self.repository.repo["name"],
+                    "number": self.pull["number"],
+                },
+            ),
+            send_fn=self.client.graphql_post,
+        )
+        resolved_review_threads = []
+        query_number = 0
+        current_query = "Q" + str(query_number)
+        async for response in responses:
+            while current_query in response:
```

Comment:
I don't get why this is a while loop. 

`for current_query, inner_response in response.items():` should work ?